### PR TITLE
[MOB-3620&3622] in-app click and close events for HTML links

### PR DESCRIPTION
**Pull request by evangreer91**
## 🔹 JIRA Ticket(s) if any

* [MOB-3620](https://iterable.atlassian.net/browse/MOB-3620)
* [MOB-3622](https://iterable.atlassian.net/browse/MOB-3622)

## ✏️ Description

This pull request includes the click and close events for the HTML links. This pull request also includes the logic for suppressing the HTML links.

### CODE REVIEW

File: ts/IterableInboxMessageDisplay.tsx [hunk 1]
```diff
@@ -86,20 +96,24 @@ const IterableInboxMessageDisplay = ({
...
+      Iterable.trackInAppClick(rowViewModel.inAppMessage, IterableInAppLocation.inbox, URL)
+
       if(URL === 'iterable://delete') {
          deleteRow(rowViewModel.inAppMessage.messageId)
          returnToInbox()
       } else if(URL === 'iterable://dismiss') {
          returnToInbox()
+         Iterable.trackInAppClose(rowViewModel.inAppMessage, IterableInAppLocation.inbox, IterableInAppCloseSource.link)
       } else {
          if(Iterable.savedConfig.urlHandler) {
```
Comment (Ayyanchira) [inline ts/IterableInboxMessageDisplay.tsx:109]: Looks like you can combine else if { } here

File: ts/IterableInboxMessageDisplay.tsx [hunk 2]
```diff
@@ -86,20 +96,24 @@ const IterableInboxMessageDisplay = ({
...
             let action = new IterableAction("openUrl", URL, "")
             let source = IterableActionSource.inApp
             let context = new IterableActionContext(action, source)
 
             Iterable.savedConfig.urlHandler(event.nativeEvent.data, context)
+            returnToInbox()
```
Comment (Ayyanchira) [inline ts/IterableInboxMessageDisplay.tsx:115]: returnToInbox() should happen in all cases right? why not just add it at the bottom common to all cases?

**Review by Ayyanchira — COMMENTED** [review]
Added some comment

File: ts/IterableInboxMessageDisplay.tsx (hunk 2 above)
Comment (evangreer91) [reply ts/IterableInboxMessageDisplay.tsx:115]: Yes. I can add returnToInbox() to the bottom for all cases.

File: ts/IterableInboxMessageDisplay.tsx (hunk 1 above)
Comment (evangreer91) [reply ts/IterableInboxMessageDisplay.tsx:109]: Yes. I will update.

File: ts/IterableInboxMessageDisplay.tsx (hunk 2 above)
Comment (Ayyanchira) [reply ts/IterableInboxMessageDisplay.tsx:115]: Or just once at the end of the method? 
Is there any case we want to exclude where returning to inbox is not expected?

File: ts/IterableInboxMessageDisplay.tsx [hunk 3]
```diff
@@ -86,20 +96,24 @@ const IterableInboxMessageDisplay = ({
...
       } else if(URL === 'iterable://dismiss') {
          returnToInbox()
+         Iterable.trackInAppClose(rowViewModel.inAppMessage, IterableInAppLocation.inbox, IterableInAppCloseSource.link)
       } else {
          if(Iterable.savedConfig.urlHandler) {
             let action = new IterableAction("openUrl", URL, "")
             let source = IterableActionSource.inApp
             let context = new IterableActionContext(action, source)
 
             Iterable.savedConfig.urlHandler(event.nativeEvent.data, context)
```
Comment (roninopf) [inline ts/IterableInboxMessageDisplay.tsx:114]: Use `URL` instead of `event.nativeEvent.data`?

Comment (evangreer91) [reply ts/IterableInboxMessageDisplay.tsx:114]: Yes. I will update.

**Pull request closed without merging**